hw-passthrough/serial/serial-over-network.md

```diff
@@ -0,0 +1,220 @@
+
+
+## Configure serial over network on EFLOW
+
+Use socat on client, we configure socat to create a pseudo-ttys as the device and assoicate the pseudo-ttys device with a network port. A pseudo-tty is like a serial port in that it has a /dev entry that can be opened by a program that expects a serial port device, except that instead of belonging to a physical serial device, the data can be intercepted by another program. The socat program intercept the traffic to/from this pseudo-ttys and relay the data for a network port.
+
+Use hub4com from com0com project on server side.  we setup hub4com to associate the network port with the physical serial port, hub4com listens to the assigned port, establish connection to client and relay data from network port to physical port.
+
+Here is an example EFLOW configuration
+
+- Host ip = **172.18.246.137**
+- Host tcp port = **5002**
+- Host physical serial port = **COM3**
+
+- Client virtual serial port name = **/dev/ttyVirtS0**
+
+
+
+### Setup Host
+Start from host can setup the host
+- Download and extract hub4com-2.1.0.0-386.zip from [com0com project](https://sourceforge.net/projects/com0com/files/hub4com/2.1.0.0/) to a local directory, e.g. c:\hub4com
+- Pick a tcp port number to associate to the serial port planned to expose.  In our example environment, we use port number 5002 and the port is associated with COM3
+    > If you have firewall enabled, explicitly configure firewall rule to allow inbound traffic for TCP port 5002.  Or run the command in next step and allow access when asked.
+- Run hub4com in server mode, open a command prompt, type `cd /d c:\hub4com` and `com2tcp.bat \\.\COM3 5002` where **COM3** is the physical serial port name and **5002** is the tcp port number.  Click 'Allow access' if Windows Defender Firewall dialog box pops up ask for port access.
+- Hub4Com starts in server mode waiting for connection.
+	
+
+### Setup Client
+- Connect EFLOW VM via Windows Admin Center or Powershell.
+- Run socat in client mode: from the WAC CLI or Powershell window, type `sudo socat pty,link=/dev/ttyVirtS0,raw,user=iotedge-user,group=dialout,mode=777 tcp:172.18.246.137:5002`.  The command create a virtual serial port (**/dev/ttyVirtS0**) and relay data between the virtual serial port and server (ip address **172.18.246.137**, port **5002**)
+
+### Verify connection
+After the setup complete, you can verify the connectino by sending data to the virtual serial port on client and see if the data is received on the servier.  The command window running hub4com will show messages when data received from or sent to the physical serial port.
+
+### Start the connection when system boot up
+You can further configure to enable the serial over network when system boot up if necessary.
+
+- Host: create a scheduled task to run hub4com.  The following powershell script example creates a scheduled task that runs when the system starts.
```

Comment:
Johnson, would it make sense to break this PS script out into a separate file from the readme.md?  I think it would make it easier for customers to consume.  In doing so, we would want the readme.md and PS file pushed down one level in the directory structure ie. /samples/serial/serial-over-network/readme.md and <script>

Reply:
yes, will create a ps script for user to use.

Reply:
ps script added